packages/adapter-databuddy/src/adapter.ts

```diff
@@ -0,0 +1,152 @@
+import type { Adapter, JsonValue } from 'flags';
+
+interface DatabuddyUser {
+  userId?: string;
+  email?: string;
+  properties?: Record<string, any>;
+}
+
+export interface DatabuddyAdapterOptions {
+  /** Databuddy Client ID */
+  clientId: string;
+  /** Databuddy API Key (optional, for server-side) */
+  apiKey?: string;
+  /** Override API URL for self-hosted users */
+  apiUrl?: string;
+  /** Environment context */
+  environment?: string;
+  /** User identification function */
+  identifyUser?: (params: { headers: any; cookies: any }) => Promise<DatabuddyUser> | DatabuddyUser;
+  /** Cache TTL in milliseconds (default: 60000) */
+  cacheTtl?: number;
+}
+
+interface FlagEvaluationResponse {
+  enabled: boolean;
+  value: JsonValue;
+  payload?: any;
+  reason: string;
+  variantName?: string;
+}
+
+/**
+ * Create a Databuddy adapter for Vercel Flags SDK
+ * @example
+ * ```typescript
+ * import { flag } from 'flags/next';
+ * import { createDatabuddyAdapter } from '@vercel/flags-adapter-databuddy';
+ *
+ * const adapter = createDatabuddyAdapter({
+ *   clientId: process.env.DATABUDDY_CLIENT_ID!,
+ *   apiKey: process.env.DATABUDDY_API_KEY,
+ *   environment: process.env.NODE_ENV || 'production',
+ * });
+ *
+ * export const showNewFeature = flag({
+ *   key: 'show-new-feature',
+ *   adapter,
+ *   defaultValue: false,
+ * });
+ * ```
+ */
+export function createDatabuddyAdapter<ValueType = JsonValue, EntitiesType = DatabuddyUser>(
+  options: DatabuddyAdapterOptions,
+): Adapter<ValueType, EntitiesType> {
+  const {
+    clientId,
+    apiKey,
+    apiUrl = 'https://api.databuddy.cc',
+    environment = 'production',
+    identifyUser,
+    cacheTtl = 60000,
+  } = options;
+
+  // Simple in-memory cache for server-side
+  const cache = new Map<string, { value: ValueType; expiresAt: number }>();
+
+  return {
+    config: {
+      reportValue: true,
+    },
+
+    origin: (key: string) => ({
+      provider: 'databuddy',
+      flagKey: key,
+      clientId,
+      environment,
+    }),
+
+    identify: async (params) => {
+      if (!identifyUser) {
+        return undefined;
+      }
+      return identifyUser(params) as EntitiesType;
+    },
+
+    decide: async ({ key, entities, headers, cookies, defaultValue }) => {
+      // Check cache first
+      const cached = cache.get(key);
```

Comment:
The cache key does not respect the `entities`, which leads to cache poisoning issues